### PR TITLE
Implement ArithmeticOps and ComparisonOps

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: fletcher/base.py [hunk 1]
```diff
@@ -310,6 +353,158 @@ def base(self):
...
+        Raises
+        ------
+        TypeError : subclass does not define reductions
+        """
+        if name == "any" and pa.types.is_boolean(self.dtype.arrow_dtype):
+            return any_op(self.data, skipna=skipna)
+        elif name == "all" and pa.types.is_boolean(self.dtype.arrow_dtype):
+            return all_op(self.data, skipna=skipna)
+        elif name == "sum" and self.dtype._is_numeric:
+            # TODO: skipna
```
Comment (fjetter) [inline fletcher/base.py]: maybe we should raise NotImplementedError?

File: tests/test_pandas_extension.py [hunk 2]
```diff
@@ -540,15 +696,136 @@ def test_reduce_series_boolean(self, data, all_boolean_reductions, skipna):
...
+        super().check_opname(s, op_name, other, exc=None)
+
+    def _compare_other(self, s, data, op_name, other):
+        self.check_opname(s, op_name, other)
+
+    def test_compare_scalar(self, data, all_compare_operators):
+        if pa.types.is_list(data.dtype.arrow_dtype):
+            pytest.xfail("pandas cannot cope with lists as scalar")
+        else:
+            # FIXME: Upstream always compares againt 0
```
Comment (fjetter) [inline tests/test_pandas_extension.py]: Is there an issue we can link?

File: tests/test_pandas_extension.py [hunk 3]
```diff
@@ -382,6 +369,8 @@ def test_combine_le(self, data_repeated):
         # Test that combine works when doing a <= (le) comparison
         # Fletcher returns 'fletcher_chunked[bool]' instead of np.bool as dtype
         orig_data1, orig_data2 = data_repeated(2)
+        if pa.types.is_list(orig_data1.dtype.arrow_dtype):
```
Comment (fjetter) [inline tests/test_pandas_extension.py]: Can this be covered by some kind of fixture? Adding it to every test is a lot of boilerplate

File: fletcher/_algorithms.py [hunk 4]
```diff
@@ -261,3 +281,187 @@ def all_op(arr: Union[pa.ChunkedArray, pa.Array], skipna: bool) -> bool:
...
+
+
+std_op = partial(pd_nanop, nanops.nanstd)
+skew_op = partial(pd_nanop, nanops.nanskew)
+kurt_op = partial(pd_nanop, nanops.nankurt)
+var_op = partial(pd_nanop, nanops.nanvar)
+median_op = partial(pd_nanop, nanops.nanmedian)
+
+
+def _in_chunk_offsets(
```
Comment (fjetter) [inline fletcher/_algorithms.py]: Didn't we implement this somewhere before?

File: fletcher/_algorithms.py [hunk 5]
```diff
@@ -261,3 +281,187 @@ def all_op(arr: Union[pa.ChunkedArray, pa.Array], skipna: bool) -> bool:
...
+) -> Tuple[List[Tuple[int, int, int]], List[Tuple[int, int, int]]]:
+    offsets_a = _calculate_chunk_offsets(a)
+    offsets_b = _calculate_chunk_offsets(b)
+    offsets = sorted(set(list(offsets_a) + list(offsets_b)))
+    in_a_offsets = _in_chunk_offsets(a, offsets)
+    in_b_offsets = _in_chunk_offsets(b, offsets)
+    return in_a_offsets, in_b_offsets
+
+
+def np_ufunc_op(op: Callable, a: Union[pa.ChunkedArray, pa.Array], b: Any):
```
Comment (fjetter) [inline fletcher/_algorithms.py]: I would feel more comfortable with this function split up and tested individually

Comment (fjetter) [inline fletcher/_algorithms.py]: Are `ufunc`s always row-wise operations?

Comment (fjetter) [reply fletcher/_algorithms.py]: I could also see some kind of dispatch implementation simplify this code, e.g.
https://github.com/dask/dask/blob/f9dd5bd3cee83f2410ef3e63c8c3d0488db70da2/dask/utils.py#L440
or
https://github.com/mrocklin/multipledispatch
or ...

File: fletcher/base.py (hunk 1 above)
Comment (xhochy) [reply fletcher/base.py]: This is actually tested and implemented, I removed the TODO and propagated the `skipna` parameter.

File: fletcher/_algorithms.py (hunk 4 above)
Comment (xhochy) [reply fletcher/_algorithms.py]: No, maybe you are referring to `_calculate_chunk_offsets` which I moved into this file?

File: fletcher/_algorithms.py (hunk 5 above)
Comment (xhochy) [reply fletcher/_algorithms.py]: I opted for https://docs.python.org/3/library/functools.html#functools.singledispatch as this is already built-in and works with type annotations. Perils of being Python 3.6+ only.

Comment (xhochy) [reply fletcher/_algorithms.py]: Yes, ufuncs are by-default row-wise. They also need to support reductions but only via `ufunc.reduce` whereas `ufunc()` is always row-wise.

File: tests/test_pandas_extension.py (hunk 3 above)
Comment (xhochy) [reply tests/test_pandas_extension.py]: Implemented a marker for skipping/xfailing tests.

File: fletcher/_algorithms.py (hunk 4 above)
Comment (fjetter) [reply fletcher/_algorithms.py]: Ok, didn't see the removal anywhere else. The logic felt familiar but I couldn't find it right away

**Pull request merged**